server/account/middleware.ts

```diff
@@ -20,12 +20,65 @@ const isCurrentSessionAccountExists = async (req: Request, res: Response, next:
   next();
 };
 
+/**
+ * Checks if the account is logged in, that is, whether the credentialId is set in session
+ */
+const isLoggedIn = (req: Request, res: Response, next: NextFunction) => {
+  if (!req.session.credentialId) {
+    res.status(401).json({
+      error: 'You must be logged in to complete this action.'
+    });
+    return;
+  }
+
+  next();
+};
+
+/**
+ * Checks if the account is signed out, that is, credentialId is undefined in session
+ */
+const isLoggedOut = (req: Request, res: Response, next: NextFunction) => {
+  if (req.session.credentialId) {
+    res.status(403).json({
+      error: 'You are already signed in.'
+    });
+    return;
+  }
+
+  next();
+};
+
+/**
+ * Checks if a name in req.body is valid, that is, it matches the name regex
+ */
+const isValidAccountName = (required: boolean) => (req: Request, res: Response, next: NextFunction) => {
+  if (!required && req.body.name === undefined) {
+    next();
+    return;
+  }
+
+  const nameRegex = /^[a-z ,.'-]+$/i;
```

Comment:
A small thing, but we talked about whether a name regex might be too restrictive (especially on foreign characters) – it might be better to check for a non-empty string instead?

Reply:
yeah this was from before that conversation. I think let's leave it for now and we can discuss briefly with the group tonight.